src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -271,8 +270,9 @@ class GrpcLb : public LoadBalancingPolicy {
 
   class Helper : public ChannelControlHelper {
    public:
-    explicit Helper(RefCountedPtr<GrpcLb> parent)
-        : parent_(std::move(parent)) {}
+    Helper(RefCountedPtr<GrpcLb> parent, bool pending_child)
```

Comment:
I think using a bool at the ctor time is not enough. If we replace the previous pending child policy with a new one, it's possible that the old one can still call `UpdateState()` and be swapped in, because the shutdown method in LB policy is in a closure scheduled by `Orphan()`. Or even worse, the new pending child policy is swapped in, then the old child policy is also swapped in and override the previous one.

This race is solved in https://github.com/grpc/grpc/pull/18073, although I'm going to abandon it.

Reply:
Thanks for pointing this out.  I think I've fixed this now.